shop/src/main/java/ro/msg/learning/shop/service/ProductCategoryService.java

```diff
@@ -0,0 +1,70 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.ProductCategory;
+import ro.msg.learning.shop.mapper.ProductCategoryMapper;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class ProductCategoryService {
+
+    public static final String THE_VALUE_IS_ALREADY_IN_THE_LIST = "The value is already in the list.";
+    public static final String ID_INVALID = "ID invalid.";
+    @Autowired
+    private ProductCategoryRepository productCategoryRepository;
+    @Autowired
+    private ProductCategoryMapper productCategoryMapper;
+
+
+    public ProductCategory createProductCategory(ProductCategory productCategory) {
+        ProductCategory productCategoryFound = productCategoryRepository.findByName(productCategory.getName()).orElse(null);
+        if (productCategoryFound != null) {
+            throw new IllegalArgumentException(THE_VALUE_IS_ALREADY_IN_THE_LIST);
```

Comment:
see above comments about this exception

Reply:
changed in EntityExistsException, also logged the message and sent the response accordingly